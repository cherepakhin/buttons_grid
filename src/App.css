
.App {
  text-align: center;
  border-radius: 0;
}

.main-app-container .list-group-item {
  display: flex;
  padding-left: 0px;
  padding-right: 12px;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.btn {
  border-radius: 0px !important;
}

.btn-primary {
   background-color: #18186A !important;
}

html body {
    background-color: #f8f9fa;
}

.main-app-container {
  padding-left: 0.5ch;
  padding-top: 0.5ch;
}

.button-div {
  background-color: #18186A;
  color: white;
  display: inline-block;
  min-width: 10ch;
  max-width: 10ch;
  text-align: center;
  line-height: 4ch;
  font-size: 2ch;
  margin-bottom: 0.5ch;
}
