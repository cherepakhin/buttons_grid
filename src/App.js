import React from 'react';
import './App.css';

class App extends React.Component { // receives props from index.js
  render() {
      return (
        //  "md-0" Margin Down (mD!) (https://react-bootstrap.github.io/docs/layout/grid)
        // this is comment OUTside Component. For comment INside Component see below.
        // path for test appView.node.props.className
        // Attention:  access to props with __THIS.props.*__
        // <NewVacancyPanel visible={this.props.visibleNewVacancyDlg}/>
        <div className="main-app-container">
          <div className="button-div mr-1 col-md-1 col-sm-1">Button1</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button2</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button3</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button4</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button5</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button6</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button7</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button8</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button9</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button10</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button11</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button12</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button13</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button14</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button15</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button16</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button17</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button18</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button19</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button20</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button21</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button22</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button23</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button24</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button25</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button26</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button27</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button28</div>
          <div className="button-div mr-1 col-md-1 col-sm-1">Button29</div>
        </div>
      );
  }
}

export default App;
