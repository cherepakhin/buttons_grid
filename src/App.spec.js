import React from "react";
import { shallow, mount } from "enzyme";
import App from "./App";
import toJson from "enzyme-to-json";


describe("<App />", () => {
  it("renders correctly", () => {
    const wrapper = mount(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});