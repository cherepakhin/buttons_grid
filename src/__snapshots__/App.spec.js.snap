// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<App>
  <div
    className="main-app-container"
  >
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button1
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button2
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button3
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button4
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button5
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button6
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button7
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button8
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button9
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button10
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button11
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button12
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button13
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button14
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button15
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button16
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button17
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button18
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button19
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button20
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button21
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button22
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button23
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button24
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button25
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button26
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button27
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button28
    </div>
    <div
      className="button-div mr-1 col-md-1 col-sm-1"
    >
      Button29
    </div>
  </div>
</App>
`;
